app/routes/planet_route.py

```diff
@@ -0,0 +1,108 @@
+from flask import Blueprint, abort, make_response, request, Response
+from app.models.planets import Planet
+from ..db import db
+from sqlalchemy import func
+
+planets_bp = Blueprint("planets_bp", __name__, url_prefix="/planets")
+
+#Function gives end user ability to search api by ID or by Name
+def validate_planet_identifier(planet_identifier):
+    # Checks if id is int or name. If id return id function if str return name function
+    if planet_identifier.isdigit():
+        return validate_planet_id(planet_identifier)
+    else:
+        return validate_planet_name(planet_identifier)
+
+# checks if user requested valid planet id and returns 404 if user enters unknown planet id
+def validate_planet_id(planet_id):
+    try:
+        planet_id = int(planet_id) 
+    except:
+        response = {"message": f"planet {planet_id} invalid"}
+        abort(make_response(response, 400))
+
+    query = db.select(Planet).where(Planet.id == planet_id)
+    planet = db.session.scalar(query)
+
+    if not planet:
+        abort(make_response({"message": f"planet {planet_id} not found"}, 404))
+    
+    return planet
+
+# checks if user requested valid planet name and returns 404 if user enter unknown planet name
+def validate_planet_name(planet_name):
+    planet_name = planet_name.lower()
+    query = db.select(Planet).where(func.lower(Planet.name) == planet_name)
+    planet = db.session.scalar(query)
+
+    if not planet:
+        abort(make_response({"message": f"planet {planet_name} not found"}, 404))
+    
+    return planet
```

Comment:
Now that we done a little more refactoring, we could move these functions into their own file to clean up our code base a little.